lib/core.js

```diff
@@ -98,6 +122,7 @@ export function core(schema, defaultTagName, caseSensitive) {
               }
             }
           } else {
+            // @ts-expect-error cannot be `HComponentProperties` because we're not on that codepath.
```

Comment:
Wouldn’t it be easier to reuse `HProperties` for props, instead of a different but similar `HComponentProperties`?
That would solve this. And probably make it easier to create a component `MyLink` which wraps `a` and internally calls `h('a', {...props, something: true})`, which would currently fail I think?